front/components/ConnectorPermissionsTree.tsx

```diff
@@ -162,14 +164,26 @@ export function DataSourceViewPermissionTreeChildren({
   viewType,
   ...props
 }: DataSourceViewPermissionTreeChildrenProps) {
-  const { nodes, isNodesLoading, isNodesError } = useDataSourceViewContentNodes(
-    {
-      dataSourceView,
-      parentId: parentId ?? undefined,
-      owner,
-      viewType,
+  const { ref, inView } = useInView();
```

Comment:
Given that we need to update 3 distinct places, shouldn't we have a wrapper around this? Also I'm not convinced by the design, the Spinner should probably be rightly aligned in the tree, WDYT?

Reply:
the idea was to have the same spinner as when "loading" is enabled :  
tree loading: 
<img width="261" alt="Screenshot 2024-09-18 at 10 03 59" src="https://github.com/user-attachments/assets/65e0a015-cbd5-460e-99e6-da74c58df9bb">

spinner at the end :   
<img width="315" alt="Screenshot 2024-09-18 at 10 03 24" src="https://github.com/user-attachments/assets/694bc3aa-e70e-46d2-b34b-4d19b307f8db">

but I can try to align with other items instead  ?

Reply:
like that : 

<img width="303" alt="Screenshot 2024-09-18 at 10 06 40" src="https://github.com/user-attachments/assets/874ece04-ec1f-4369-a602-f2b6fc7f50e3">

Reply:
>Given that we need to update 3 distinct places, shouldn't we have a wrapper around this?  

I hesitated to put it in swr hook, but imo this should only handle swr here - I can also move to another hook

Reply:
moved logic to a dedicated component

Reply:
🙏